include/openthread/platform/settings.h

```diff
@@ -73,6 +73,7 @@ enum
     OT_SETTINGS_KEY_SRP_SERVER_INFO      = 0x000d, ///< The SRP server info (UDP port).
     OT_SETTINGS_KEY_BR_ULA_PREFIX        = 0x000f, ///< BR ULA prefix.
     OT_SETTINGS_KEY_BR_ON_LINK_PREFIXES  = 0x0010, ///< BR local on-link prefixes.
+    OT_SETTINGS_KEY_BR_ID                = 0x0011, ///< Unique BR ID.
```

Comment:
How is the new key going be to used? 
Should we define related type and methods for it in `settings.hpp/cpp`?

Reply:
The new key is used for storing the Border Router ID in ot-br-posix https://github.com/openthread/ot-br-posix/pull/1832.

Reply:
Yes. I saw the other PR after this :)

I feel may be better to add OT APIs to (maybe under `otBrAgent`?) to get/save this key? (suggested the same there)

I am suggesting this because we want all settings keys defined here to be used/managed by OT core. If we add a key that is used by `ot-br-posix` (and not used by OT core itself), it opens the door for other keys to be added in the future that are tied to some specific platform/radio and/or app use-case.

Reply:
Thanks @abtink! Updated this PR to provide API for retrieving the Border Agent/Router ID. PTAL